hnsecure/src/main/java/ca/bc/gov/hlth/hnsecure/Route.java

```diff
@@ -38,102 +41,113 @@
-                    .setBody(simple(SampleMessages.e45ResponseMessage))
-	            
-	            //sending message to HIBC for ENROL
-	            .when(simple("${in.header.messageType} == {{hibc-r50-endpoint}}"))
-	                .log("the HIBC endpoint (${in.header.messageType}) is reached and message will be dispatched to message queue(ENROL).")
-                    .setBody(simple(SampleMessages.r50ResponseMessage))
-	            
-	            //others sending to JMB
-	            .otherwise()
-                    .log("the JMB endpoint is reached and message will be dispatched to JMB!!")
-                    .setBody(simple(SampleMessages.r03ResponseMessage))
-            .end();
-
-    }
+	private String pharmanetCert;
+
+	private static final String pharmanetCertPassword = System.getenv("PHARMANET_CERT_PASSWORD");
+
+	private static final String pharmanetUser = System.getenv("PHARMANET_USER");
+
+	private static final String pharmanetPassword = System.getenv("PHARMANET_PASSWORD");
+
+	private static ApplicationProperties properties;
+
+	private Validator validator;
+
+	@Override
+	public void configure() {
+		init();
+
+		onException(CustomHNSException.class).process(new ExceptionHandler()).handled(true);
+
+		onException(org.apache.http.conn.HttpHostConnectException.class)
+				.setHeader(Exchange.HTTP_RESPONSE_CODE, constant(500)).handled(true).setBody(constant(""))
+				.log("Failed to connect remote server");
+
+		onException(ValidationFailedException.class).log("Validation exception response: ${body}").handled(true)
+				.id("ValidationException");
+
+	
+		setupSSLConextPharmanetRegistry(getContext());
+		String pharmNetUrl = String.format(pharmanetUri
+				+ "?bridgeEndpoint=true&sslContextParameters=#%s&authMethod=Basic&authUsername=%s&authPassword=%s",
+				SSL_CONTEXT_PHARMANET, pharmanetUser, pharmanetPassword);
+		String basicToken = buildBasicToken(pharmanetUser, pharmanetPassword);
+		log.info("Using pharmNetUrl: " + pharmNetUrl);
+
+		String isFileDropsAllowed = properties.getValue(IS_FILEDDROPS_ALLOWED);
+		
+		from("jetty:http://{{hostname}}:{{port}}/{{endpoint}}").routeId("hnsecure-route")
+				.onCompletion()//It must have route scope and called only once.
+				.choice().when(header("isFileDropsAllowed").isEqualToIgnoreCase(Boolean.TRUE)).bean(ResponseFileDropGenerater.class).id("ResponseFileDropGenerater")
+				.to("log:foo").end().end()
+				.log("HNSecure received a request").setHeader("isFileDropsAllowed")
+				.simple(isFileDropsAllowed)
+				// Extract the message using custom extractor and log
+				.setBody().method(new FhirPayloadExtractor()).log("Decoded V2: ${body}")
+				//Added wireTap for asynchronous call to filedrop request
+				.wireTap("direct:start").choice().when(header("isFileDropsAllowed").isEqualToIgnoreCase(Boolean.TRUE))
+				.bean(RequestFileDropGenerater.class).id("V2FileDropsRequest").end()
```

Comment:
It looks like the request file drop isn't actually a part of the wiretap call. The wiretapped route here is just doing logging